app/main.py

```diff
@@ -1,5 +1,82 @@
+from __future__ import annotations
+
+
 class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        if coords is None:
+            coords = [0, 0]
+        self.name = name
+        self.weight = weight
+        self.coords = coords
+
+    def go_forward(self, step: int = 1) -> list:
+        x, y, *z = self.coords
+        y += step
+        self.coords = [x, y, *z]
+        return self.coords
+
+    def go_back(self, step: int = 1) -> list:
+        x, y, *z = self.coords
+        y -= step
+        self.coords = [x, y, *z]
+        return self.coords
+
+    def go_right(self, step: int = 1) -> list:
+        x, y, *z = self.coords
+        x += step
+        self.coords = [x, y, *z]
+        return self.coords
+
+    def go_left(self, step: int = 1) -> list:
+        x, y, *z = self.coords
+        x -= step
+        self.coords = [x, y, *z]
+        return self.coords
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
+        super().__init__(name, weight, coords)
```

Comment:
You could use `or` to define the argument that you pass to the `super().__init__`
```suggestion
        super().__init__(name, weight, coords or [0, 0, 0])
```